src/test/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiActionTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.dlic.rest.api;
+
+import org.apache.hc.core5.http.Header;
+import org.apache.http.HttpStatus;
+import org.junit.Test;
+
+import org.opensearch.security.test.helper.rest.RestHelper;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringContains.containsString;
+
+public class AuthFailureListenersApiActionTest extends AbstractRestApiUnitTest {
+
+    private static final Header ADMIN_FULL_ACCESS_USER = encodeBasicHeader("admin_all_access", "admin_all_access");
+    private static final Header USER_NO_REST_API_ACCESS = encodeBasicHeader("admin", "admin");
+
+    @Test
+    public void testForbiddenAccess() throws Exception {
+        setupWithRestRoles();
+
+        rh.sendAdminCertificate = false;
+        RestHelper.HttpResponse getSettingResponse = rh.executeGetRequest(
+            "/_plugins/_security/api/authfailurelisteners",
+            USER_NO_REST_API_ACCESS
+        );
+        assertThat(getSettingResponse.getBody(), getSettingResponse.getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
+        RestHelper.HttpResponse updateSettingResponse = rh.executePutRequest(
+            "/_plugins/_security/api/authfailurelisteners/test",
+            "{\"type\":\"ip\",\"allowed_tries\":10,\"time_window_seconds\":3600,\"block_expiry_seconds\":600,\"max_blocked_clients\":100000,\"max_tracked_clients\":100000}",
+            USER_NO_REST_API_ACCESS
+        );
+        assertThat(getSettingResponse.getBody(), updateSettingResponse.getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
+        RestHelper.HttpResponse deleteSettingResponse = rh.executeDeleteRequest(
+            "/_plugins/_security/api/authfailurelisteners/test",
+            USER_NO_REST_API_ACCESS
+        );
+        assertThat(getSettingResponse.getBody(), updateSettingResponse.getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
+    }
+
+    @Test
+    public void testFullAccess() throws Exception {
```

Comment:
Can we add a test, that has a minimal payload? i.e. just include `{\"type\":\"ip\"}` and verify that it will create an auth failure listener with the default values?

Default values are here: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/impl/v7/ConfigV7.java#L230-L234

Will the integer parsing allow negative values? Can we prohibit negative ints?

Reply:
Done in: [2a8c237](https://github.com/opensearch-project/security/pull/4641/commits/2a8c237f25239167cf854bc50143f9f7f85a4af9). I don't think we need to handle negative values/that is overkill since they could always just put negative values in the yaml right? We can bring that up as a separate issue to validate that doesn't break anything but it feels out of scope for this PR.